test/common/crypto/utility_test.cc

```diff
@@ -109,7 +109,8 @@ TEST(UtilityTest, TestVerifySignature) {
   EXPECT_EQ(false, result.result_);
   EXPECT_EQ("unknown is not supported.", result.error_message_);
 
-  PublicKeyObject* empty_crypto = new PublicKeyObject();
+  PublicKeyObject* empty_crypto = static_cast<PublicKeyObject*>(malloc(sizeof(PublicKeyObject)));
```

Comment:
@PiotrSikora thoughts here?

Reply:
The issue is with mixing `new`+`free`, instead of using either `new`+`delete` or `malloc`+`free`. 

`PublicKeyObject` is an Envoy thing, not BoringSSL, and I imagine that `new`+`delete` is the preferred way to handle this in C++ world, but it's up to you, really.

Reply:
If the mismatched free is in Envoy code and not boringssl code, yes, please fix this without placement new.

/wait

Reply:
This PR has a simple goal of to fix the mismatch problem in the test code. 

To change the free in verifySignature to delete may require some ssl/lua lib knowledge (which I hope someone else could fix it if new+delete is desired in this lib):
The problem comes from that verifySignature actually deletes (may or may not, depends on if the lua_touserdata contains a non-null ::pkey_) a referenced parameterr inside its body. the referenced "key" is actually from a raw pointer from lua_touserdata, I think the verifySignature actually takes ownership of that piece of ram, but I am not 100% sure of it. 

in lua_filter.cc:
                                                                                                                                                                                                                                                  
```
446   // Step 2: get key pointer                                                                                                                                                                                                                     
447   auto ptr = lua_touserdata(state, 3);                                                                                                                                                                                                           
448                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
....
....
458   // Step 5: verify signature                                                                                                                                                                                                                    
459   auto crypto = reinterpret_cast<Common::Crypto::CryptoObject*>(ptr);                                                                                                                                                                            
460   auto& crypto_util = Envoy::Common::Crypto::UtilitySingleton::get();                                                                                                                                                                            
461   auto output = crypto_util.verifySignature(hash, *crypto, sig_vec, text_vec); 

```

in verifySignature: 

```
                                                                                                                                                                                                               
 55   // Step 3: initialize EVP_DigestVerify                                                                                                                                                                                                         
 56   auto pkey_wrapper = Common::Crypto::Access::getTyped<Common::Crypto::PublicKeyObject>(key);                                                                                                                                                    
 57   EVP_PKEY* pkey = pkey_wrapper->getEVP_PKEY();                                                                                                                                                                                                  
 58                                                                                                                                                                                                                                                  
 59   if (pkey == nullptr) {                                                                                                                                                                                                                         
 60     free(pkey_wrapper);                                                                                                                                                                                                                          
 61     return {false, "Failed to initialize digest verify."};                                                                                                                                                                                       
 62   }                                                                                                                                                                                                                                              
 63                                                                                                                                                                                                                                                  
 64   int ok = EVP_DigestVerifyInit(ctx.get(), nullptr, md, nullptr, pkey);                                                                                                                                                                          
 65   if (!ok) {                                                                                                                                                                                                                                     
 66     return {false, "Failed to initialize digest verify."};                                                                                                                                                                                       
 67   }                
```

I am not sure if there is a leak if we dont make sure release pkey_wrapper inside verifySignature(), or we should simple not free/delete at all inside the verifySignature() function as a reference parameter implies the ownership is not held by the function.

But again, I don't know what does EVP_DigestVerifyInit do or lua_filter.cc manages the ram or not... 

Please let me know if the free() statement can be removed from verifySignature? 
or VerifySignature should take a unique_ptr instead of a reference?
or someone else please make the change, and let me only fix the mismatched free-new problem that's in the utility_test.cc :P

Reply:
ugh I don't think the free should be in verifySignature, and actually this may crash. so deleting the free statement is right thing to do.